src/ctapipe/tools/calculate_pixel_stats.py

```diff
@@ -0,0 +1,194 @@
+
+    name = "ctapipe-calculate-pixel-statistics"
+    description = "Perform statistics calculation for pixel-wise image data"
+
+    examples = """
+    To calculate statistics of pixel-wise image data files:
+
+    > ctapipe-calculate-pixel-statistics --TableLoader.input_url input.dl1.h5 --output_path /path/monitoring.h5 --overwrite
+
+    """
+
+    allowed_tels = Set(
+        trait=CInt(),
+        default_value=None,
+        allow_none=True,
+        help=(
+            "List of allowed tel_ids, others will be ignored. "
+            "If None, all telescopes in the input stream will be included."
+        ),
+    ).tag(config=True)
+
+    input_column_name = Unicode(
+        default_value="image",
+        allow_none=False,
+        help="Column name of the pixel-wise image data to calculate statistics",
+    ).tag(config=True)
+
+    output_table_name = Unicode(
+        default_value="statistics",
+        allow_none=False,
+        help="Table name of the output statistics",
+    ).tag(config=True)
+
+    output_path = Path(
+        help="Output filename", default_value=pathlib.Path("monitoring.h5")
+    ).tag(config=True)
+
+    overwrite = Bool(help="Overwrite output file if it exists").tag(config=True)
+
+    aliases = {
+        ("i", "input_url"): "TableLoader.input_url",
+        ("o", "output_path"): "PixelStatisticsCalculatorTool.output_path",
+    }
+
+    flags = {
+        "overwrite": (
+            {"PixelStatisticsCalculatorTool": {"overwrite": True}},
+            "Overwrite existing files",
+        ),
+    }
+
+    classes = [
+        TableLoader,
+    ] + classes_with_traits(PixelStatisticsCalculator)
+
+    def setup(self):
+        # Read the input data with the 'TableLoader'
+        self.input_data = TableLoader(
+            parent=self,
+        )
+        # Check that the input and output files are not the same
+        if self.input_data.input_url == self.output_path:
+            raise ToolConfigurationError(
+                "Input and output files are same. Fix your configuration / cli arguments."
+            )
+        # Load the subarray description from the input file
+        subarray = SubarrayDescription.from_hdf(self.input_data.input_url)
+        # Get the telescope ids from the input data or use the allowed_tels configuration
+        self.tel_ids = (
+            subarray.tel_ids if self.allowed_tels is None else self.allowed_tels
+        )
+        # Initialization of the statistics calculator
+        self.stats_calculator = PixelStatisticsCalculator(
+            parent=self, subarray=subarray
+        )
+
+    def start(self):
+        # Iterate over the telescope ids and calculate the statistics
+        for tel_id in self.tel_ids:
+            # Read the whole dl1 images for one particular telescope
+            dl1_table = self.input_data.read_telescope_events_by_id(
+                telescopes=[
+                    tel_id,
+                ],
+                dl1_images=True,
```

Comment:
These are all configuration options. By hardcoding them here as kwargs, you prevent users from overriding them via the config.

Reply:
@maxnoe fixed